mem_context.go

```diff
@@ -66,15 +66,16 @@ type MemUsageContext struct {
 	memoryLimit                    uint64
 }
 
+var latestMemUsageContext *MemUsageContext
```

Comment:
[q] can this package var be concurrently accessed?

Reply:
I'm afraid the answer is yes and something I plan to tackle (didn't want to do it yet until I test this locally first before making all of this concurrent-safe).

Reply:
Will resolve for now since I added a Mutex here, for any comment feel free to leave them in the new version.